### ISSUE TITLE
[java] JUnit 4 false positives for JUnit 5 tests

### ISSUE THREAD

**Issue by AdamCarroll** [pmd/pmd#940]
**Affects PMD Version:** 

6.1.0

**Rule:**

* JUnit4TestShouldUseBeforeAnnotation
* JUnit4TestShouldUseAfterAnnotation
* JUnit4TestShouldUseTestAnnotation

**Description:**

JUnit 4 rules cause false positive in JUnit 5 tests.  It looks like the new annotations for JUnit 5 aren't being handled as they should be.

**Code Sample demonstrating the issue:**

The file [`FileFormatType`](https://github.com/VocabHunter/VocabHunter/blob/d6428810ee89707a17a4b3d9d59d33bf7c8bf0e6/core/src/test/java/io/github/vocabhunter/analysis/core/FileToolTest.java) on the project [VocabHunter](https://github.com/VocabHunter/VocabHunter) shows this problem.  To work around the problem, the following lines have been added to [ruleset.xml](https://github.com/VocabHunter/VocabHunter/blob/b979a1b25a079076887ab88b7fc03be1dfe74e72/config/pmd/ruleset.xml#L15-L17):
```
<rule ref="category/java/bestpractices.xml">
    <exclude name="JUnit4TestShouldUseBeforeAnnotation"/>
    <exclude name="JUnit4TestShouldUseAfterAnnotation"/>
    <exclude name="JUnit4TestShouldUseTestAnnotation"/>
</rule>
```

To reproduce the problem, remove the above workaround line and then launch the following:
```
$ ./gradlew clean :core:pmdTest
```

**Running PMD through:** *Gradle*

**Comment by vovkss** [comment]
I've faced the same problem after upgrade to JUnit5 and PMD 6.5. 

Here a pull request proposal which would solve the problem: https://github.com/pmd/pmd/pull/1256/files
